### PR TITLE
Improve testing for cluster deletion

### PR DESCRIPTION
**Pull request by aclevername**
### Description

I thought I had found an issue where we were issuing kubernetes calls when the API was unavailable, but after adding the tests it appears not. 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by michaelbeaumont — APPROVED** [review]
> I thought I had found an issue where we were issuing kubernetes calls when the API was unavailable, but after adding the tests it appears not.

Did any behavior change in the end?

**Comment by aclevername** [comment]
> > I thought I had found an issue where we were issuing kubernetes calls when the API was unavailable, but after adding the tests it appears not.
> 
> Does this PR change any behavior in the end?

I don't believe so

File: pkg/actions/cluster/owned_test.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+		ranDeleteClusterTasks = false
+		ctl = &eks.ClusterProvider{Provider: p, Status: &eks.ProviderStatus{}}
+	})
+
+	Context("when the cluster is operable", func() {
+		It("deletes the cluster", func() {
+			//mocks are in order of being called
+			p.MockEKS().On("DescribeCluster", mock.MatchedBy(func(input *awseks.DescribeClusterInput) bool {
+				Expect(*input.Name).To(Equal(clusterName))
+				return true
```
Comment (cPu1) [inline pkg/actions/cluster/owned_test.go:57]: Generally, assertions should not be added to the matcher predicate function. The library handles presenting meaningful errors when the predicate fails. 

```suggestion
				return *input.Name == clusterName
```

**Review by cPu1 — APPROVED** [review]
LGTM! I only have one comment around using assertions in the `mock.MatchedBy` predicate function.

**Pull request merged**